src/main/java/uk/gov/companieshouse/search/api/controller/upsert/UpsertCompanyController.java

```diff
@@ -1,25 +1,31 @@
 package uk.gov.companieshouse.search.api.controller.upsert;
 
+import static uk.gov.companieshouse.search.api.logging.LoggingUtils.COMPANY_NAME;
+import static uk.gov.companieshouse.search.api.logging.LoggingUtils.COMPANY_NUMBER;
+import static uk.gov.companieshouse.search.api.logging.LoggingUtils.UPSERT_COMPANY_NUMBER;
+import static uk.gov.companieshouse.search.api.logging.LoggingUtils.getLogger;
+
 import java.util.HashMap;
 import java.util.Map;
 
 import javax.validation.Valid;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
-import uk.gov.companieshouse.search.api.logging.LoggingUtils;
 import uk.gov.companieshouse.search.api.mapper.ApiToResponseMapper;
 import uk.gov.companieshouse.search.api.model.response.ResponseObject;
+import uk.gov.companieshouse.search.api.model.response.ResponseStatus;
 import uk.gov.companieshouse.search.api.service.upsert.UpsertCompanyService;
 
 @RestController
-@RequestMapping(value = "/upsert-company")
+@RequestMapping(value = "/alphabetical-search/companies/{company_number}")
```

Comment:
I think with the update to the endpoint being the same as what is present in the `AlphabeticalSearchController` we should delete the `UpsertCompanyController` and move it to `AlphabeticalSearchController`

Reply:
Depends if the idea is to mimic this upsert functionality for the dissolved companies though doesn't it?

Reply:
I still think it would require a different endpoint for dissolved and would be better placed in the corresponding controller. With the endpoint now being uniform with the addition of the path parameter - I think its better suited and follows restful standards if it was in the `AlphabeticalSearchController`